### ISSUE TITLE
Zorro ELRS Date/time occassionly reset

### ISSUE THREAD

**Issue by pfeerick** [EdgeTX/edgetx#2187]
I have the same problem with the Radiomaster Zorro ELRS version. Date/time reset every reboot sometimes to 2000, sometimes to 1970 year. EdgeTX 2.7.1 and nightly 2.8.0 the same behavior.

_Originally posted by @Roman-Golod in https://github.com/EdgeTX/edgetx/issues/1588#issuecomment-1206107884_

**Comment by Roman-Golod** [comment]
It doesn't happen, when I pull off the battery (I use the external 2s battery) without switching off Zorro. Time/date saves in that case. Maybe I can make any another experiments to help solving this problem?

**Comment by nrlourenzo** [comment]
Is there any update on this? I have exactly the same issue, date resets to date 2000-1-1

**Comment by Roman-Golod** [comment]
I received new quartz and capacitors from Radiomaster, replaced it, but it didn't help. Though date/time resetting now happens less often. Maybe l'll try to do it ond more time. So, i think, it's hardware related problem.

**Comment by TerraMotoEU** [comment]
Got a new Zorro 4-in-1 that came with EdgeTX 2.7.1 and had the same problem: Date and time reset after reboot. Issue still persisting after update to 2.8. Mostly resets to yr 2000 but sometimes to 1970.

**Comment by pfeerick** [comment]
You'll find when it resets to 1970, most likely it will be 2000 on the next boot... since 1970 is the default date of the RTC, and is from before when OTX/ETX was first written, so is then set to 2000. The problem is finding out *why* it is resetting... Companion doesn't have anything to do with setting the time, although some people seem to see it happen after a sync with Companion. I've not yet experienced this myself - probably because I never initially set the time on my Zorros, but once set they seem to have stayed set. 

This almost feels like the backup RTC battery is flat, but there must be something more to this as this would be an easily fixed occurrence (and wouldn't be evident until after several months to a couple of years of ownership)

**Comment by gagarinlg** [comment]
I think someone should take a look at the RTC supply voltage line with a decent oscilloscope when switching on and off and when un-/-plugging USB.

@Risto do you have a Zorro?

**Comment by Roman-Golod** [comment]
10 days ago I rebuilt the SD content and never touched the data port since then. I thought, the problem was gone, but yesterday it happened again. The clock battery is OK (3,3v).

**Comment by TerraMotoEU** [comment]
Ah the battery! Mine's at 3.24v. No idea how low they're allowed to go, though.

**Comment by GavoGavo** [comment]
Got mine today.

ELRS version on 2.7, and experiencing the same issue. Have noticed it go to both 1970 and 2020.

I have mostly been using it plugged in to the PC for flying Sim. 

I wonder if the process of plugging in and out is related to the cause?

It's very frustrating.

**Comment by TerraMotoEU** [comment]
Contemplating whether I'd send it back to The Netherlands for replacement or not. After ordering from Radiomaster EU, PostNL took forever to even move my order. If I was sure it's something that can be fixed via fw update, I'd hold on to it and wait.

Aside from messing up file names (logs, model backups...) is there any other risk coming along with this issue, that we should be aware of, or is that all there is?

**Comment by GavoGavo** [comment]
Were these all new radios?

Did anyone else notice sticker residue on the cathode/positive side of the batteries, when they removed the clear-and-red battery tab that breaks the circuit in transit?

I noticed when I removed mine that quite a lot of residue was left over. I used the tab to remove the remaining residue but I tried charging the batteries in my XTAR charger and noticed they were not reliably making a circuit. I tried getting more residue off and they seemed kind of OK after that. 

I tried cleaning them futher just now.. some residue did seem to come off and the charger now has no problem detecting them. I wonder if there might be a physical component here... poor contact perhaps. I don't think so because I would think that any drop in contact would switch the device off but perhaps it's worth a thought?

If you do try and clean the terminals, be really, really careful and read up about how to do it before you do. These batteries can be dangerous and though they're protected, a short could be catastrophic. 

I do note that I have been unable to repeat the problem so far today anyway, and also upgraded firmware to 3.8.

I am going to keep trying to reproduce the problem. If it happens again I will lodge a warranty request with Radiomaster to see what they say. Hopefully this is just a small batch of units. I do wonder if we all got devices from the same batch maybe.

**Comment by andreo32** [comment]
I got an answer, when I addressed the same question to radiomaster support:
`This is a known issue, a STM related bug that we've communicated to edge TX, who should be addressing the issue with a future update.`

**Comment by andreo32** [comment]
Hi!
There's a guy in [RC groups who explaining RTC reset](https://www.rcgroups.com/forums/showpost.php?p=50318429&postcount=1210) issue by replacing original RTC battery to new - a little bigger that original.
Maybe is just a to weak battery problem...
Br,

**Comment by TheIsotopes** [comment]
@pfeerick Is there anything news on this topic?

I'm having the same problem and it's frustrating.

**Comment by Dopossum** [comment]
I had the same problem, Time and Date resetting to 2000 or 1970.
 I have the Radiomaster Zorro Elrs V2.0  running under EdgeTX 2.7. 
In the menu SYS/Radio Setup , where also Date and Time are set 
there is a Checkbox option " Adjust RTC" . I enabled it.
I have rebooted several times since and Date and Time where everytime correct.

**Comment by raphaelcoeffic** [comment]
“Adjust RTC only make sense when a GPS is connected that is then used to fetch the current time. Your issue is most probably related to the RTC itself, and it’s supply power. I believe we have fixed such an issue that related to some brownout level. At best you should update to 2.8, which has the fix included.

**Comment by KC748** [comment]
Hi everyone. I'm the guy who replaced the RTC battery with a bigger one mentioned above by andreo32.

My Zorro is on v2.8.0 and just as pfeerick has described above ( and in #1588), the RTC will reset to 1970 if the Power button is pressed while the main battery is removed. And then reset to 2000 on the next reboot. The RTC retains it's setting in subsequent boot cycles after that.

It doesn't seem like this issue is fixed yet?

> You'll find when it resets to 1970, most likely it will be 2000 on the next boot... since 1970 is the default date of the RTC, and is from before when OTX/ETX was first written, so is then set to 2000. ...
> 
> This almost feels like the backup RTC battery is flat, but there must be something more to this as this would be an easily fixed occurrence (and wouldn't be evident until after several months to a couple of years of ownership)

**Comment by kf4mot** [comment]
> It doesn't seem like this issue is fixed yet?

It's most definitely not fixed.

**Comment by chemist132** [comment]
Same here. i had mine for 1.5 years without trouble and now the date and time are buggy(: No solution yet.

**Comment by mixMugz** [comment]
Any news about bug? Zorro 2.8.1 EdgeTX still reset date/time after reboot(poweroff)

**Comment by KC748** [comment]
Mine doesn't reset after a reboot. It only resets at certain instances as described above.
If it is resetting after a reboot, the coin cell may be going flat. The cr1220 are small and only last about 1.5-2 years.

I don't know if this is related, but after replacing the cr1220 with a brand new Duracell cr2032, the beginning 3.33v has dropped to 3.08v in 2.5 months. This drop seems excessive. 

![Menu50](https://user-images.githubusercontent.com/39738927/225484676-a7395255-3ecd-4d97-9735-28b91a97c921.jpg)


![RCT Batt](https://user-images.githubusercontent.com/39738927/225483862-f529ae36-e721-46b7-9a0a-f6ef1585ccf9.jpg)

**Comment by mixMugz** [comment]
3.08V is not enough?

**Comment by pfeerick** [comment]
I would think so, it just seems unusually quick for it to drop from 3.33v to 3.08v - although 3v is the nominal voltage. 

It should be a lithium primary cell (thus non-rechargable).

**Comment by mixMugz** [comment]
So. Another test.
Bought the new CR1220. Now voltage show 3.3 volts.
Seting date/time.
Step one - power off button. Date still correct after power on.
Step two - power off by removing battery. Date still correct after power on.
Step three - connect wia usb (joystik hid) then power off. Date reset after power on.

Now every time i power off radio - date will be reset. Sometimes date will keep, but reset after second/third reboot

**Comment by KC748** [comment]
While doing some searches in OpenTx for RTC issues, I came across this one from way back in 2015. It looks like this RTC reset issue has been around for quite a while and on various radios.
https://github.com/opentx/opentx/issues/2626 
In this issue thread, Mike Blandford has a possible code fix, 
https://github.com/opentx/opentx/issues/2626#issuecomment-128933531
and there is also a hardware fix (for the Taranis X9E) by replacing a capacitor with a larger value capacitor. 

Any chance of implementing either of these solutions for the Zorro?

**Comment by pfeerick** [comment]
The comment that closed that issue is rather telling also... 
> It's an hardware problem, please contact FrSky (the 100uF is not enough, they will use a 470uF instead)

The code that Mike Blandford commented on is already in OTX and ETX, and has been for about four years - was added as part of https://github.com/opentx/opentx/pull/6501 - and is used by basically all the B&W radios that have a soft power button (rather than switch). 

So it seems for that particular radio, it was a hardware issue through and through, and was also very intermittent, so really does fit the pattern we're seeing here. So the question now is probably going to be how to confirm it is an issue related to a capacitor value being too low (and which one)?

Nice bit of investigation... could be getting closer! :)

**Comment by AlxSlk686** [comment]
Hello! I'm having the same problem with my zorro radio which is using firmware 2.8.1 I found a clear dependency on resetting the date and time. If you turn off the radio immediately after exiting the system menu or from it, the date and time are reset, if you exit the menu and wait a few seconds, the date remains unchanged.

**Comment by KC748** [comment]
> The comment that closed that issue is rather telling also...
> 
> > It's an hardware problem, please contact FrSky (the 100uF is not enough, they will use a 470uF instead)
> 
> The code that Mike Blandford commented on is already in OTX and ETX, and has been for about four years - was added as part of [opentx/opentx#6501](https://github.com/opentx/opentx/pull/6501) - and is used by basically all the B&W radios that have a soft power button (rather than switch).
> 
> So it seems for that particular radio, it was a hardware issue through and through, and was also very intermittent, so really does fit the pattern we're seeing here. So the question now is probably going to be how to confirm it is an issue related to a capacitor value being too low (and which one)?
> 
> Nice bit of investigation... could be getting closer! :)

Thanks for the additional information.

I've reread #2626, and according to projectkk2glider, in his post, https://github.com/opentx/opentx/issues/2626#issuecomment-128792600,  the additional 470 uF capacitor is added in parallel to Vcc (3.3v) and GND.

>add additional capacitor in parallel somewhere between Vcc and GND lines,
>
>That trace is your Vcc line and should have 3.3V when the radio is powered on. The GND is on the copper fill surrounding it

I've located the 3.3v regulator (A1117B 3.3) on my Zorro and did some continuity tests and came up with the diagram below. Conveniently, there is a 3.3v and GND pads close to the regulator. 

Following projectkk2glider's instructions, I'm about to add a 470 uF capacitor on those pads, Does anyone see any harm in doing that?



![A1117B](https://user-images.githubusercontent.com/39738927/226222607-21d910be-d51d-4732-80ac-f087891cfa43.JPG)

**Comment by KC748** [comment]
Ok. I gave it a day for someone to talk me out of it...  Crickets. So here it is. :)

Following projectkk2glider's instructions, I added a 470 uF 10v capacitor (recommanded is 6v or higher) on those pads (3.3v and GND) and tacked it down with a bit of hot glue over a piece of electrical tape for insulation. 

On my Zorro (v2.2), a sure way to cause the RTC to reset is to press the Power button when the main batteries are out. The Power button would briefly flash blue. The RTC would reset to 1970, then to 2000 on the subsequent reboot.

So far, I haven't been able to cause the RTC to reset after adding the 470 uF capacitor. I'm quite satisfied with the result.

Tested:
* Power button press after removing the main battery, briefly as well as holding for more that 5 minutes.
* Flash firmware (v2.8.1) with USB connected and via SD card in bootloader mode. (Power On with horizontal trim to center)
* Flash firmware via USB with power Off.
* USB connected in Joystick mode and Storage.

Even after bricking the radio a couple of time, by pulling the USB before ejecting the Zorro's drive from the computer, the RTC did not reset after unbricking.

Disclaimer: 
Do the mod at your own risk. I'm not an expert. :)
If anyone tries this mod, some feedback would be nice.

[In the pics below, note the offset placement of the capacitor, to clear the gimbal sensor housing and wires.]

![2-470uF10v](https://user-images.githubusercontent.com/39738927/226708706-8662944e-45c9-4146-8f05-f73a26859093.JPG)
![3-Offset](https://user-images.githubusercontent.com/39738927/226708741-47c13698-6a24-4762-9066-fe2d216727af.JPG)

**Comment by mixMugz** [comment]
Good news! I need to try with this stuff
![image](https://user-images.githubusercontent.com/17923916/226722063-76e0b3c5-8fd7-435d-8075-33340de2f646.png)

**Comment by KC748** [comment]
It's curious that the tantalum capacitor above the 3.3v regulator is a 47uF capacitor (476J) and not a 470uF (477j). I'm wondering if Radiomaster screwed up and put the wrong value there.

**Comment by pfeerick** [comment]
> On my Zorro (v2.2), a sure way to cause the RTC to reset is to press the Power button when the main batteries are out. The Power button would briefly flash blue. The RTC would reset to 1970, then to 2000 on the subsequent reboot.

Oh, thanks for that - I think I have had this happen myself then... so will try to reproduce this, and then try the mod also if it happens repeatably for me. I'm now also curious as to whether the ELRS zorro is more prone to this than the MPM one, as it *could* be a little more power-hungry at startup and be causing a brownout, hence why the bigger cap needed. Anyway, something to pass on the RM for them to investigate also.

**Comment by AlxSlk686** [comment]
I installed back the firmware from RM 2.7.0 and all the problems with resetting the time finally disappeared. No matter how I try to reproduce this problem. All problems start on firmware 2.7.1 So it still looks like this is a software problem and not hardware

**Comment by richardclli** [comment]
It is interesting tough change between 2.7.0 and 2.7.1 are mainly UI related.  Cannot spot any change that may affect the RTC.

**Comment by pfeerick** [comment]
Indeed... IIRC the brownout PR was part of 2.7.0, so it won't be that. and rolling both FW and bootloarder back to 2.7.0 don't make any difference to either my MPM or ELRS Zorros for the "power on when batteries removed" triggering of the RTC being reset.

**Comment by AlxSlk686** [comment]
> Интересно, что изменения между 2.7.0 и 2.7.1 в основном связаны с пользовательским интерфейсом. Не удается обнаружить какие-либо изменения, которые могут повлиять на RTC.

I myself cannot understand why this is so. My zorro was originally 2.7.0, then I started updating it and there was a problem. I reverted to 2.7.0 Now it's been another day since my last post and the time hasn't reset as usual. I periodically try to cause the problem with different actions but everything is in order. Mystic))))

**Comment by TheIsotopes** [comment]
The capacitor mod helped me ... since @KC748 posted it i have it in action and i've seen no clock reset.

**Comment by KC748** [comment]
That’s good news! Could I ask which version of Zorro you have? 

There are couple of reports on RCG, both ELRS versions, that rests at every boot. It would be nice to give them some hope. Mine is the 4in1 and  it has been rock solid since adding the cap.

**Comment by TheIsotopes** [comment]
>  Could I ask which version of Zorro you have?

4in1,  Zoro Board v1.9

**Comment by Roman-Golod** [comment]
Great work, thanks! I've just added 1000 uF capacitor and no resets after some on/off cycles! Zorro ELRS board 2.2

**Comment by mixMugz** [comment]
So. I can confirm. The mod is working.
Putting 470uF / 6.3V on board and I no longer have any case with time/date reset.! 

![image](https://user-images.githubusercontent.com/17923916/230786130-aa2617de-cf70-4e3b-a363-b3f59847101d.png)
![image](https://user-images.githubusercontent.com/17923916/230786141-9478c3d4-3f2f-4869-acc5-b9e38c857176.png)

**Comment by bbolgradov** [comment]
Excellent, thanks! Mod works on Zorro Board v2.5 with ELRS.

**Comment by dominikm23** [comment]
There is a socket on the other side of the board which is directly connected to the 3.3V trace. I just connected a matching plug to a 470uF capacitor and plugged it in. No need to solder or disassemble the circuit board.

I have not had a single date reset since this change and it used to do it pretty much every time I turned it on.

![zorro-3v](https://user-images.githubusercontent.com/131966485/234840905-d9afc293-74ae-4990-8310-bf3577a16de7.jpg)

**Comment by Berlinologe86** [comment]
Raeiomaster says it is a bug on edgetx? But why is it only on This Radio 😔
![IMG_9829](https://user-images.githubusercontent.com/51958903/236621530-e350fb2f-1a1a-4ea5-b4ad-a2599f2257a6.jpeg)

can a edgetx dev confirm This ?

**Comment by chemist132** [comment]
I have a Tx16S with edge and the same problemen. I orderbedrag a 470 muF condensator.Op 6 mei 2023 13:33 schreef Berlinologe86 ***@***.***>:
Raeiomaster says it is a bug on edgetx? But why is it only on This Radio 😔

can a edgetx dev confirm This ?

**Comment by Shpek** [comment]
I have a BETAFPV Lite Radio 3 Pro 2.8.2 with exactly the same problem - date is set to 01-01-1970 on every reset. RTC battery is 3.1V. I also tried with the battery from my TX16S (which is working fine) - no effect. Tried the cap mod with a 470uF electrolytic and 100uF tantalum - still resetting. Can't downgrade to 2.7.0 since it doesn't have support for Lite Radio 3 Pro.

**Comment by Shpek** [comment]
#

> > Tried the cap mod with a 470uF electrolytic and 100uF tantalum - still resetting.
> 
> I tried 470uF with no effect, but 1000uF helped me. Try to increase capacity.

1000uF made it reset to 01-01-2000. Tried with 4700uF - same thing.

**Comment by pfeerick** [comment]
> Radiomaster says it is a bug on edgetx? 
> can a edgetx dev confirm This ?

That's interesting... to the best of my knowledge is it related to a design choice made by RM to ensure you don't get any mid-air power-offs due to oxidisation of the battery contacts. There are two electrolytic capacitors that are holding enough charge to cause the RTC circuit to stop working properly, hence the time/date reset. Adding more capacitance is probably allowing the MCU to start enough to realise it's in a brownout condition and stop itself properly. Removing them would also have the same effect, but then you'd want to make sure you take care of the battery contacts to prevent intermittent power. 

Other than that, this is a nearly impossible issue to fault-find, as outside of the Zorro and an unusual/extreme sequence of events, there does not seem to be any pattern that allows us to reliably reproduce the fault on our end.

**Comment by Thomas-Cee** [comment]
So @KC748 , could I remove the surface mount 47μF and remount a larger surface mount cap like @mixMugz pointed to? Does it need to be an electrolytic? 

![Cap Mod](https://github.com/EdgeTX/edgetx/assets/38837288/56db80b7-e5fa-4da7-8a9d-d48e9013a688)

**Comment by KC748** [comment]
I'm really not the right person to ask, to be honest.
My sense is that the tantalum would be ok instead of an electrolytic. But Personally, I'd just add it to the 3.3v/GND pads. It's way less risky. 
Hopefully someone will chime in with a more definitive answer. Or otherwise, you're welcome to blazing the trail on this one. :)

**Comment by zatalian** [comment]
I noticed that if I keep pressing the power off button longer on turn off, time is not reset. After the 4 dots the sleeping face appears and stays on as long as i keep pressing the power button. If I do that for a few seconds, the clock is not reset.

This made me wonder if maybe the haptic feedback between the dots and the sleeping face is causing the reset. I turned haptic feedback off and for now. No resets since...

**Comment by kf4mot** [comment]
I replaced the pathetic little coin cell in my Zorro 4 months ago. Now it's down to 3 volts again and I got another RTC reset.
Now I soldered in a 2032 size cell. Time will tell if it stays set any longer, but with roughly 8X capacity hopefully it last a couple years.

**Comment by MAcevedo03** [comment]
Below is a reply I received from Marketing at Radiometer after I wrote to them about my resetting issue with the 4 in 1 RC.
It's sounds like you can send the RC back, however I'm not sure the "new" RC won't have the same issue.

Thanks for your contact. On the SYS-Hardware page of the remote control, you can see the RTC Battery voltage. When it is lower than 2V, the calendar and time may not be saved correctly, indicating that it needs to be replaced. You can also tick the Check RTC on this page. When the button battery is out of power, the remote control will pop up a prompt message.

**Comment by lightbulbjim** [comment]
Since adding a 470uF capacitor across the 3.3V rail a few months ago I haven't had any more resets. Not sure whether to put the fault down to poor design or cheap (out of spec) components.

**Comment by CountParadox** [comment]
I just received my transparent zorro, I am experiencing the RTC reset issue also, I will try the cap mod and see if it helps. Just wanted to let people know its still an issue on new units :)

**Comment by Thomas-Cee** [comment]
I wonder how a larger cap install, and how a larger 2032 button cell that @kf4mot referenced interface with eachother? 
Interesting that they both seem to cure the reset issue.
Does the cap, provide a brief burst of current instead of being drawn from the battery, causing the battery to last longer?
That's interesting. I do wonder what at a circuit level is happening in there...

**Comment by kf4mot** [comment]
Without schematics we're all just kind of speculating. I've added caps and a bigger RTC battery because I'm a belt and suspenders sort of guy. I don't see that either _conclusively_ solves the problem though unfortunately.
I've also considered adding a ceramic cap parallel with the RTC battery since coin cells can't handle high current draw. If the RTC ever needs a higher current pulse that could help. I have no evidence that's a issue though either. The larger CR2032 would handle higher current pulses better than the stock RTC battery on it's own by being larger.

**Comment by TrevorCrouch** [comment]
I've had quite a few Frsky, TBS RX which have worked fine. Splashed out on the Zorro Max a few weeks ago and have the rest issue.  It's annoying for sure and the above remedies appear to 'possibly work for a while'. My question is, does it effect anything if it does reset back to yr 2k or 1970??

**Comment by TrevorCrouch** [comment]
> It will affect logging and backups. Maybe the last modified time of files if you use the SD sync feature in Companion.
> 
> Adding the capacitor has fixed it reliably on my two Zorros.

Thanks LBJ, did you install just the 470uf?

**Comment by TrevorCrouch** [comment]
Can you confirm the dimensions of the 470uF 10v you used please Jim

**Comment by lightbulbjim** [comment]
Not without opening up a radio, no. But you can find specs online [easily enough](https://www.digikey.com/en/products/filter/aluminum-electrolytic-capacitors/58?s=N4IgjCBcoEwAwBYCcVQGMoDMCGAbAzgKYA0IA9lANogIDscABAK0BiIAuqQA4AuUIAVQB2ASx4B5TAFlC2fAFcAToRABfUvFopoIDJBwES5KuEYA1Dtz6RBoidNkLla1aqA).

**Comment by TrevorCrouch** [comment]
Not a problem, I didn't know if you had more to hand. As I'm buying some and the 25v is a bit big, I'll grab som 10v's. Thanks for your input.

**Comment by lightbulbjim** [comment]
You should be able to find a 6.3V version which will be even smaller and easier to tuck out of the way.

**Comment by sremick** [comment]
@lightbulbjim The location you used for the capacitor appears quite different than other peoples' photos. Is it really identical to the one near the voltage regulator?

**Comment by lightbulbjim** [comment]
It just needs to be across the 3.3V rail and ground. Physical location doesn’t matter.

**Comment by gagarinlg** [comment]
I would liek to close this issue. It is not SW related and we can not do anything here within EdgeTX

**Comment by corneliusmunz** [comment]
@gagarinlg the only weirded thing is, that I never ever had that behaviour on the same Radio Hardware (Zorro 4in1 with Ranger Nano external Module) with the 2.7.0 version. So I am not sure, if it is really HW dependent.

**Comment by W9CVO** [comment]
> @gagarinlg the only weirded thing is, that I never ever had that behaviour on the same Radio Hardware (Zorro 4in1 with Ranger Nano external Module) with the 2.7.0 version. So I am not sure, if it is really HW dependent. 

I've been having the issue with both the 2.7.0 and 2.9.0 firmware in my Zorro 4in1 with Ranger Nano as well.  Trying out the cap mod today to see if it helps.

**Comment by sremick** [comment]
> I used a 10V since it's what I had to hand, but anything over 3.3V will be fine. Larger voltage ratings mean the capacitor will be physically larger so it might be a challenge to fit in the case.
> 
> It's on the left in the picture, near the coin cell. I did both my Zorros the same.
> 
> ![IMG_2096](https://user-images.githubusercontent.com/4312404/258768229-4673ed8f-0562-49d3-bbd4-3a84a9977ede.jpeg)

I soldered mine to the same pads, but I instead bent the capacitor over the left edge of the PCB so it actually sits on the other side of the PCB. I did it this way to reduce the amount of exposed leads on the capacitor and chances that it might short against something else on the PCB.

**Comment by taavituisk** [comment]
Holding the power button a few seconds before letting go has helped to avoid resetting the rtc with my Zorro. If i let go as soon as the the shutoff finishes will reset the rtc quite reliably.

**Comment by zatalian** [comment]
> Holding the power button a few seconds before letting go has helped to avoid resetting the rtc with my Zorro. If i let go as soon as the the shutoff finishes will reset the rtc quite reliably.

This no longer works for me with edgetx 2.9.1. Maybe this can be a clue for the devs to solve the problem? Whatever changed between 2.8.4 and 2.9.1, maybe you can do the opposite to solve the problem?
Or maybe i should no longer postpone the capacitor solution...

**Comment by CountParadox** [comment]
470u 10v cap on 3.3v rail did not resolve the reset issue for me :(

**Comment by Roman-Golod** [comment]
> 470u 10v cap on 3.3v rail did not resolve the reset issue for me :( 

Try 1000

**Comment by Quas7** [comment]
I also had it not reliable with 470uF close to the coin cell. Therefore, I added a 10uF SMD capacitor on the not-coin-cell-side to the 3.3V+GND pads. That fixed the issue.

Because of that, I wanted to note, that the capacitor needs to be close to the actual RTC and that the RTC is likely not right next to the battery. That could also be the reason, why we see this issue as a long thin trace over the full board between battery and RTC part might have to much resistance and inductance to buffer the switch off event.

In short - just install another 470uF capacitor on the other side as well, if it is not working. ;)

**Comment by JCHPennington** [comment]
Hi, I have just bought a new Zorro 4in1, and noticed the same reset issue. Software updated to 2.9.2  Raised it with Radiomaster support and they are recommending a main board swap out with no instructions on what to do or specialist tools.  The unit is new and hardly used.  Was not expecting to have to replace the main board so early in the product life cycle. Any advice appreciated.

**Comment by W9CVO** [comment]
> Hi, I have just bought a new Zorro 4in1, and noticed the same reset issue. Software updated to 2.9.2  Raised it with Radiomaster support and they are recommending a main board swap out with no instructions on what to do or specialist tools.  The unit is new and hardly used.  Was not expecting to have to replace the main board so early in the product life cycle. Any advice appreciated. 

Are they shipping you a new board or do they expect you to buy it yourself?  It appears to be only Phillips head screws to remove the rear cover and the board itself.  I started a ticket with RM myself as well and am waiting for a reply.  I was going to replace the off brand button cell RTC battery today in hopes of that fixing it but Amazon sent me the wrong battery.

**Comment by JCHPennington** [comment]
Hi W9CVO

Many thanks for your email, RM are planning to send me a new main board but
I am pushing back as do not expect to have to do the work on such a new
unit. Also what happens if unit does not work post replacement.

I have had issues with calibration values not holding as well.

I shall let you know how I get on. The radio is still very new and not
actually used on an aircraft which has flown as I've been learning edge Tx

Julian

**Comment by KC748** [comment]
A member on RcGroups reported that the replacement main board he received from Radiomaster solved the resetting RTC problem on his Zorro.
https://www.rcgroups.com/forums/showthread.php?4027391-RadioMaster-ZORRO-is-here&postid=51714643

If the new board doesn’t fix the problem, then you could try and add a capacitor as many in this tread have done. (470uF-1000uF, 6v or higher). There are numerous attached photos in this thread that shows where the capacitor should be installed. 

In any case, photos of the new board would be of interest to see what has changed. If you are able, please post some pictures. Both sides of the new board.

**Comment by JCHPennington** [comment]
Hi,

Many thanks for that it is appreciated.  I am just waiting for Radiomaster to replace the main board.  I have requested that the board replacement is done at regional repair facility as it looks very detailed repair despite the offer to send it to myself to action.  I don’t really want to 'open up’ the radio as putting it back together is not without risk of none-functioning.  Although for others it is not considered to be a risk.
 

Hopefully this will resolve the issue from everyone’s perspective.


Regards

Julian

**Comment by gsimonnet** [comment]
Radiomaster sent me a new main board under warranty, which I just replaced, and so far it fixes the problem.
The old board was version 2.2, the new one version 2.5. The electrolytic capacitors are the same on both.
See picture for comparison.
![RMZorro](https://github.com/EdgeTX/edgetx/assets/42216699/bf24aa21-6b3f-40b0-b3cf-7fa333baa653)

**Comment by mikecarr** [comment]
> Radiomaster sent me a new main board under warranty, which I just replaced, and so far it fixes the problem. The old board was version 2.2, the new one version 2.5. The electrolytic capacitors are the same on both. See picture for comparison. ![RMZorro](https://private-user-images.githubusercontent.com/42216699/298361108-bf24aa21-6b3f-40b0-b3cf-7fa333baa653.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDgwNTUyMTYsIm5iZiI6MTcwODA1NDkxNiwicGF0aCI6Ii80MjIxNjY5OS8yOTgzNjExMDgtYmYyNGFhMjEtNmIzZi00MGIwLWIzY2YtN2ZhMzMzYmFhNjUzLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjE2VDAzNDE1NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJiYzRiMjAwNjU3MWVhZmNhODIwOTliOTM2YTJhZjAyMjBiZjg1NDhlNGI3MmQyN2Y4ODQ4MDg1MzRmY2FmNWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.A-Bknw3ddYu_2aAIXcOzjyEt0XIZyYV-ZAZR6R5h2ts)

I am also getting a new board, will update once I get it. They said they would send after the New Year

**Comment by gsimonnet** [comment]
> > Radiomaster sent me a new main board under warranty, which I just replaced, and so far it fixes the problem. The old board was version 2.2, the new one version 2.5. The electrolytic capacitors are the same on both. See picture for comparison. ![RMZorro](https://private-user-images.githubusercontent.com/42216699/298361108-bf24aa21-6b3f-40b0-b3cf-7fa333baa653.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDgwNTUyMTYsIm5iZiI6MTcwODA1NDkxNiwicGF0aCI6Ii80MjIxNjY5OS8yOTgzNjExMDgtYmYyNGFhMjEtNmIzZi00MGIwLWIzY2YtN2ZhMzMzYmFhNjUzLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjE2VDAzNDE1NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJiYzRiMjAwNjU3MWVhZmNhODIwOTliOTM2YTJhZjAyMjBiZjg1NDhlNGI3MmQyN2Y4ODQ4MDg1MzRmY2FmNWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.A-Bknw3ddYu_2aAIXcOzjyEt0XIZyYV-ZAZR6R5h2ts)
> 
> I am also getting a new board, will update once I get it. They said they would send after the New Year

Unlike me, remember to reflash the new board to match your microSD card before using...

**Comment by sremick** [comment]
I'm going to swap the motherboard on my Radiomaster Zorro so I want to ensure I back up everything so I can restore it to the way it was before after the work is done. Because of the nature of this, it's imperative that I ensure I've backed up everything 100% first since it won't be easy to back and get anything I missed.

Going by this:

https://oscarliang.com/setup-zorro/#Backup-and-Restore-of-Radio-Settings

...is it sufficient to back up the "Radio Settings" and "Model Profiles" or is there anything else?

**Comment by gsimonnet** [comment]
> I'm going to swap the motherboard on my Radiomaster Zorro so I want to ensure I back up everything so I can restore it to the way it was before after the work is done. Because of the nature of this, it's imperative that I ensure I've backed up everything 100% first since it won't be easy to back and get anything I missed.
> 
> Going by this:
> 
> https://oscarliang.com/setup-zorro/#Backup-and-Restore-of-Radio-Settings
> 
> ...is it sufficient to back up the "Radio Settings" and "Model Profiles" or is there anything else?

I usually backup the content of the entire microSD card since it is not that big. After the swap, just reuse your old SD card, but reflash the board to match the last version of EdgeTc installed (i.e. the last bin file flashed from the FIRMWARE folder).

**Comment by pfeerick** [comment]
Reading the radio with Companion and saving the result to an `.etx` file should be sufficient for recovery (of anything firmware version related). Alternately, a copy of the `MODEL` and `RADIO` folders. But a copy of the entire card certainly don't hurt. Plus it is pretty quick and painless to maintain if you use the Sync option in Companion... it only copies the changes on subsequent runs.

**Comment by strauss115** [comment]
I also got a replacement board under warranty from Radiomaster. With the new board the date/time reset issue is fixed. Interestingly, the old board version was also 2.5. So it seems that is a general issue not related to a specific board version.

**Comment by sloped-soarer** [comment]
Hi.
Not sure if I have the same issue.
However I can state with confidence that the RTC resets on my Zorro when it is powered down and the top USB port is connected to a computer (forcing the DFU mode).
Zorro is new. Lithium coin cell was also new and it's voltage was checked before testing.

**Comment by JCHPennington** [comment]
Hi yes its the same issue I have.

**Comment by sloped-soarer** [comment]
Hi. I got this response from Radiomaster support.

"
Yes, the RTC clock will reset when entering DFU mode, this is a normal phenomenon, it will not reset without entering DFU.
When you shut down or take out the 18350 battery and put it back in later, if the time is not reset, it means that the remote control is normal, it will reset only in DFU mode, and will not reset when not using DFU.
"
I would like to know what processor is used in the Zorro and why entering DFU resets the RTC ?.

Also my Zorro randomly resets the RTC through normal power on  / off cycles.

**Comment by lightbulbjim** [comment]
I’m afraid that response from RadioMaster comes across as wilfully ignorant. 🙄On 16 Jul 2024, at 21:50, sloped-soarer ***@***.***> wrote:﻿
Hi. I got this response from Radiomaster support.
"
Yes, the RTC clock will reset when entering DFU mode, this is a normal phenomenon, it will not reset without entering DFU.
When you shut down or take out the 18350 battery and put it back in later, if the time is not reset, it means that the remote control is normal, it will reset only in DFU mode, and will not reset when not using DFU.
"
I would like to know what processor is used in the Zorro and why entering DFU resets the RTC ?.
Also my Zorro randomly resets the RTC through normal power on  / off cycles.

**Comment by JCHPennington** [comment]
Hi this is problem and now my zorro will not power up at all despite a full
charged batteries.

More challenges and not happy

Regards

Julian

**Comment by agnauck** [comment]
On my Zorro the RTC issue was resolved by deleting radio.yaml from the SDcard. After radio.yaml got recreated by EdgeTX the issue was resolved. But I don't have a backup of the old "corrupted" yaml file for diagnosis.

**Comment by JCHPennington** [comment]
Hi Alexander, thats great news. I am still quite new to EdgeTx could you
explain how you did it?

Many thanks

Julian

**Comment by CountParadox** [comment]
Can someone with the issue post a copy of their radio.yaml file from the SD card?On 29 Oct 2024 7:14 am, JCHPennington ***@***.***> wrote:
Hi Alexander, thats great news. I am still quite new to EdgeTx could you

explain how you did it?



Many thanks



Julian



On Mon, 28 Oct 2024, 21:00 Alexander Gnauck, ***@***.***>

wrote:

**Comment by agnauck** [comment]
> Hi Alexander, thats great news. I am still quite new to EdgeTx could you explain how you did it? Many thanks Julian

The file is on your SDCard at /RADIO/radio.yml

**Comment by agnauck** [comment]
sorry for not coming back earlier herre. It was a hardware issue and the capacitor mod fixed it.

**Comment by JCHPennington** [comment]
Hi, dont worry about it. I received a new circuit board but it looks as if
it was not modificatied to install a new capacitor. So I am left with the
same issue.

Regards

Julian

**Comment by 3djc** [comment]
Most boards don't require mod at all, only a few exhibit this issue

**Comment by JCHPennington** [comment]
Okay well it appears that I have one of those boards! Heigh Ho....

**Comment by sercona** [comment]
I updated my zorro to the latest and I still have the date reset issue.

what hw mod is there for this?  I see mention of a cap but I dont see any details about it.

I'm ok installing a cap if I need to.  is there a photo or something to go by?

**Comment by agnauck** [comment]
@sercona please read all messages in this issue here. There are many hints and pictures of installing a capacitor.

When I had an affected Zorro a while ago I fixed it like described here, because it works without soldering on the main board with just a connector:
https://fpv-community.de/threads/radiomaster-zorro.86854/page-4#post-1105628

**Comment by JCHPennington** [comment]
Hi could you send me a copy of the photo you found so I can have a look and
tey myself. Still concerned that this is known issue and Radiomaster offer
no support or modification kit. Heigh ho...

Julian
***@***.***
WhatsApp


Movil Espagne: 692 503 196
Mobile UK: 07540744320

**Comment by sercona** [comment]
so far, this HAS fixed my zorro.  I've tried my best to power 'surge' with on/off presses and that holding cap, well.... it holds ;)  that was the fix.  thanks!
